SecretSanta/test/SecretSanta.Api.Tests/Controllers/GroupControllerTests.cs

```diff
@@ -126,41 +127,41 @@ public void UpdateGroup_ReturnsUpdatedGroup()
         [TestMethod]
         [DataRow(-1)]
         [DataRow(0)]
-        public void DeleteGroup_RequiresPositiveId(int groupId)
+        public async Task DeleteGroup_RequiresPositiveId(int groupId)
         {
             var service = new Mock<IGroupService>(MockBehavior.Strict);
             var controller = new GroupsController(service.Object, Mapper.Instance);
 
-            var result = controller.Delete(groupId);
+            IActionResult result = await controller.Delete(groupId);
 
             Assert.IsTrue(result is BadRequestObjectResult);
         }
 
         [TestMethod]
-        public void DeleteGroup_ReturnsNotFoundWhenTheGroupFailsToDelete()
+        public async Task DeleteGroup_ReturnsNotFoundWhenTheGroupFailsToDelete()
         {
             var service = new Mock<IGroupService>();
             service.Setup(x => x.DeleteGroup(2))
-                .Returns(false)
+                .Returns(Task.FromResult(false))
                 .Verifiable();
             var controller = new GroupsController(service.Object, Mapper.Instance);
 
-            var result = controller.Delete(2);
+            IActionResult result = await controller.Delete(2);
 
             Assert.IsTrue(result is NotFoundResult);
             service.VerifyAll();
         }
 
         [TestMethod]
-        public void DeleteGroup_ReturnsOkWhenGroupIsDeleted()
+        public async Task DeleteGroup_ReturnsOkWhenGroupIsDeleted()
         {
             var service = new Mock<IGroupService>();
             service.Setup(x => x.DeleteGroup(2))
-                .Returns(true)
+                .Returns(Task.Run(()=>true))
```

Comment:
You don't want to use a `Task.Run` here. This will execute the lambda on a background thread which is not what you want. 
```suggestion
                .ReturnsAsync(true)
```